tensorflow/contrib/metrics/python/ops/metric_ops_test.py

```diff
@@ -6660,5 +6661,194 @@ def test2dWeightedValues_placeholders(self):
+
+  def testVars(self):
+    metrics.cohen_kappa(
+        predictions=array_ops.ones((10, 1)),
+        labels=array_ops.ones((10, 1)),
+        num_classes=2)
+    _assert_metric_variables(self, (
+        'cohen_kappa/total_po:0',
+        'cohen_kappa/total_pe_row:0',
+        'cohen_kappa/total_pe_col:0',))
+
+  def testMetricsCollection(self):
+    my_collection_name = '__metrics__'
+    kappa, _ = metrics.cohen_kappa(
+        predictions=array_ops.ones((10, 1)),
+        labels=array_ops.ones((10, 1)),
+        num_classes=2,
+        metrics_collections=[my_collection_name])
+    self.assertListEqual(ops.get_collection(my_collection_name), [kappa])
+
+  def testUpdatesCollection(self):
+    my_collection_name = '__updates__'
+    _, update_op = metrics.cohen_kappa(
+        predictions=array_ops.ones((10, 1)),
+        labels=array_ops.ones((10, 1)),
+        num_classes=2,
+        updates_collections=[my_collection_name])
+    self.assertListEqual(ops.get_collection(my_collection_name), [update_op])
+
+  def testValueTensorIsIdempotent(self):
+    predictions = random_ops.random_uniform(
+      (10, 1), maxval=3, dtype=dtypes_lib.int64, seed=1)
+    labels = random_ops.random_uniform(
+      (10, 1), maxval=3, dtype=dtypes_lib.int64, seed=2)
+    kappa, update_op = metrics.cohen_kappa(labels, predictions, 3)
+
+    with self.test_session() as sess:
+      sess.run(variables.local_variables_initializer())
+
+      # Run several updates.
+      for _ in range(10):
+        sess.run(update_op)
+
+      # Then verify idempotency.
+      initial_kappa = kappa.eval()
+      for _ in range(10):
+        self.assertAlmostEqual(initial_kappa, kappa.eval(), 5)
+
+  def testBasic(self):
+    confuse_matrix = np.array([
+      [9, 3, 1],
+      [4, 8, 2],
+      [2, 1, 6]])
+    labels, predictions = self._confuse_matrix_to_samples(confuse_matrix)
+
+    dtypes = [dtypes_lib.int32, dtypes_lib.int64,
+              dtypes_lib.float32, dtypes_lib.float64]
+    shapes = [(len(labels,)),  # 1-dim
+              (len(labels), 1)]  # 2-dim
+    weights = [None, np.ones_like(labels)]
+
+    for dtype in dtypes:
+      for shape in shapes:
+        for weight in weights:
+          with self.test_session() as sess:
+            predictions_tensor = constant_op.constant(
+                np.reshape(predictions, shape), dtype=dtype)
+            labels_tensor = constant_op.constant(
+                np.reshape(labels, shape), dtype=dtype)
+            kappa, update_op = metrics.cohen_kappa(
+                labels_tensor, predictions_tensor, 3, weights=weight)
+
+            sess.run(variables.local_variables_initializer())
+            self.assertAlmostEqual(0.45, sess.run(update_op), 2)
+            self.assertAlmostEqual(0.45, kappa.eval(), 2)
+
+  def testBasic2(self):
+    labels = np.random.randint(0, 4, size=(100, 1))
+    predictions = np.random.randint(0, 4, size=(100, 1))
+    expect = sk_metrics.cohen_kappa_score(labels, predictions)
```

Comment:
I would prefer if we don't depend on sk_metrics, since the calculation is relatively simple if the data are small. This would also solve the problem you found importing sklearn.

Reply:
I think sk_metrics give us more secure, however it is OK to calculate the `expect` by ourselves.